From c9fe230b2842c5db32863c6e5aa2b2ce9c5679fd Mon Sep 17 00:00:00 2001
From: Yi Sun <yi.sun@intel.com>
Date: Thu, 2 Jan 2025 11:15:54 +0800
Subject: [PATCH] iaa_test: Use syscall write to submit descriptor
'Content-type:text/plain'

Align with the kernel change, write the descriptor to cdev instead of
enqcmd by default when submit shared work queue descriptor.

Do the same change for iaa_test as the what dsa_test, which is reviewed
and merged.

Signed-off-by: Yi Sun <yi.sun@intel.com>
---
 test/iaa_test.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/iaa_test.c b/test/iaa_test.c
index 5854c64495f3..d42fa73bbb06 100644
--- a/test/iaa_test.c
+++ b/test/iaa_test.c
@@ -28,6 +28,7 @@ static void usage(void)
 	"-n <number of descriptors> ;descriptor count to submit\n"
 	"-t <ms timeout> ; ms to wait for descs to complete\n"
 	"-v              ; verbose\n"
+	"-u              ; use ENQCMD to submit descriptor\n"
 	"-h              ; print this message\n");
 }
 
@@ -581,7 +582,7 @@ int main(int argc, char *argv[])
 	int dev_wq_id = ACCTEST_DEVICE_ID_NO_INPUT;
 	unsigned int num_desc = 1;
 
-	while ((opt = getopt(argc, argv, "w:l:f:1:2:3:a:m:o:b:c:d:n:t:p:vh")) != -1) {
+	while ((opt = getopt(argc, argv, "w:l:f:1:2:3:a:m:o:b:c:d:n:t:p:vuh")) != -1) {
 		switch (opt) {
 		case 'w':
 			wq_type = atoi(optarg);
@@ -624,6 +625,9 @@ int main(int argc, char *argv[])
 		case 'v':
 			debug_logging = 1;
 			break;
+		case 'u':
+			force_enqcmd = 1;
+			break;
 		case 'h':
 			usage();
 			exit(0);
-- 
2.48.0

